pkg/controller/master/virtualmachine/vmi_controller.go

```diff
@@ -35,6 +41,71 @@ type VMIController struct {
 	recorder            record.EventRecorder
 }
 
+// SyncHarvesterVMILabelsToPod ensures that all Harvester labels (i.e. those
+// with the harvesterhci.io/ prefix) from the VirtualMachineInstance are synced
+// to the Pod
+func (h *VMIController) SyncHarvesterVMILabelsToPod(_ string, vmi *kubevirtv1.VirtualMachineInstance) (*kubevirtv1.VirtualMachineInstance, error) {
+	if vmi == nil || vmi.DeletionTimestamp != nil {
+		return vmi, nil
+	}
+
+	logrus.Debugf("Syncing labels %v for VMI %v to Pod", vmi.Labels, vmi.Name)
+
+	harvesterVMILabels := map[string]string{}
```

Comment:
we should skip the VMI if the maintenance strategy is `Migrate`. This will allow kubevirt to just handle the migration of VM's by watching for nodes being placed into maintenance mode via its own watchers. 
We do not need to do anything specific for virt-launcher pods in this case.

The additional benefit of this will be that we will not be patching the pod labels for every launcher pod. It is very likely that `Maintenance Stratergy` is only used by a subset of VM's among a large number of end user clusters. This greatly reduces the number of reconciles / update requests to the apiserver for the pods.

Reply:
This controller is not just for the maintenance-mode functionality. It syncs all labels with the `harvesterhci.io` prefix, which may come in handy later on, e.g. when we need to match Pods to their Rancher downstream clusters, or when we need to label Pods of node-pools, so load-balancers can identify them.

I'm also not a huge fan of implementing this functionality in a controller, but I don't see many other options.